playbooks/roles/monitoring-user-defined-projects/tasks/main.yml

```diff
@@ -0,0 +1,171 @@
+---
+
+## Enabling monitoring to user-defined-projects
+
+- set_fact:
+    default_namespace: "{{ '-'.join(['user','namespace',random_digit]) }}"
+    default_service_monitor_name: "{{ '-'.join(['project','service-monitor',random_digit]) }}"
+  vars:
+    random_digit: "{{ 100 | random }}"
+
+- name: Create a new namespace
+  k8s:
+    state: present
+    definition:
+      apiVersion: v1
+      kind: Namespace
+      metadata:
+        name: "{{ user_defined_namespace | default(default_namespace, true) }}"
+
+- name: Deploy a sample app
+  k8s:
+    state: present
+    definition: 
+      apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        labels:
+          app: "{{ user_defined_app_name | default('prometheus-example-app', true) }}"
```

Comment:
Why are we using default() everywhere? We should make use of the variables in `defaults/main.yml`.

Reply:
Implemented fact variables and assigned the role variables to the respective parameter along with the default value to avoid the use of `default()` in the play tasks.